### PR TITLE
Add support for spaces and users in GrantToken

### PR DESCRIPTION
**Pull request by kleewho**
feat: Add support for spaces and users permissions in GrantToken

feat: Add new Get/SetUserId and NewConfigWithUserId factory method for Config. Deprecate UUID

### CODE REVIEW

File: grant_token_request.go [hunk 1]
```diff
@@ -21,6 +21,12 @@ type grantTokenBuilder struct {
 	opts *grantTokenOpts
 }
 
+type SpaceId string
+
+type grantTokenSumBuilder grantTokenBuilder
+
+type grantTokenObjectsBuilder grantTokenBuilder
```
Comment (phairow) [inline grant_token_request.go:28]: Do we really want to use the `Objects` qualifier in the name? I thought we were moving away from `Objects`

Comment (phairow) [reply grant_token_request.go:28]: I know I was out last week so correct me if I'm wrong. What are other SDKs doing with this naming conflict?

File: grant_token_request.go [hunk 2]
```diff
@@ -21,6 +21,12 @@ type grantTokenBuilder struct {
 	opts *grantTokenOpts
 }
 
+type SpaceId string
+
+type grantTokenSumBuilder grantTokenBuilder
```
Comment (phairow) [inline grant_token_request.go:26]: This seems like we are introducing a new qualifier `SUM` for the VSP stuff. Is this something the team is doing across all SDKs?

**Review by phairow — COMMENTED** [review]
My only concern with this PR is the names `Objects`, `Objects V2` and `Sum` which I don't think we want to introduce into the codebase.

**Review by phairow — APPROVED** [review]
Given the team has standardized on the names for builders using `objects` and `entities`, this looks good to go.

**Comment by kleewho** [comment]
@client-engineering-bot release as v7.1.0

**Pull request merged**